src/util/permissions.ts

```diff
@@ -1,249 +1,328 @@
 import { cacheHandlers } from "../api/controllers/cache.ts";
-import { Guild, Role } from "../api/structures/mod.ts";
+import { Channel, Guild, Member, Role } from "../api/structures/mod.ts";
 import { botID } from "../bot.ts";
-import { Permission, Permissions, RawOverwrite } from "../types/mod.ts";
-
-/** Checks if the member has this permission. If the member is an owner or has admin perms it will always be true. */
-export async function memberIDHasPermission(
-  memberID: string,
-  guildID: string,
-  permissions: Permission[],
+import { Errors, Permission, Permissions } from "../types/mod.ts";
+
+async function getCached(
+  table: "guild",
+  key: string | Guild,
+): Promise<Guild>;
+async function getCached(
+  table: "channel",
+  key: string | Channel,
+): Promise<Channel>;
+async function getCached(
+  table: "member",
+  key: string | Member,
+): Promise<Member>;
+async function getCached(
+  table: "guild" | "channel" | "member",
+  key: string | Guild | Channel | Member,
 ) {
-  const guild = await cacheHandlers.get("guilds", guildID);
-  if (!guild) return false;
-
-  if (memberID === guild.ownerID) return true;
-
-  const member = (await cacheHandlers.get("members", memberID))?.guilds.get(
-    guildID,
-  );
-  if (!member) return false;
+  const cached = typeof key === "string"
+    ? // @ts-ignore -
```

Comment:
```suggestion
    ?
```

Reply:
This ignore comment is necessary either way.